home/views.py

```diff
@@ -363,94 +364,62 @@ def allocationForm(request):
     """
     caterer_list = Caterer.objects.filter(visible=True).all()
     alloc_form = AllocationForm.objects.filter(active=True).last()
-    try:
-        if not alloc_form:
-            raise Exception("Form is closed for now")
+    text = ""
+    message = ""
+
+    if not alloc_form:
+        message = "Form is closed for now"
+    else:
         student = Student.objects.filter(email__iexact=str(request.user.email)).last()
         if not student:
-            raise Exception(
-                "Signed in account can not fill the allocation form. Please inform the dining Office to add your email ID to the database"
-            )
-        text = ""
-        message = ""
-        if (alloc_form.start_time and alloc_form.start_time > now()) or (
+            message = "Signed in account cannot fill the allocation form. Please inform the dining Office to add your email ID to the database"
+        elif (alloc_form.start_time and alloc_form.start_time > now()) or (
             alloc_form.end_time and alloc_form.end_time < now()
         ):
-            raise Exception("The Form is closed for now")
-        elif Allocation.objects.filter(
-            email=student, period=alloc_form.period
-        ).exists():
-            raise Exception(
-                "You have filled the form for this period. Please visit the profile page after the allocation process is completed to check your allocated caterer"
-            )
+            message = "The Form is closed for now"
+        elif Allocation.objects.filter(email=student, period=alloc_form.period).exists():
+            message = "You have filled the form for this period. Please visit the profile page after the allocation process is completed to check your allocated caterer"
         elif request.method == "POST" and request.user.is_authenticated:
             period_obj = alloc_form.period
-            high_tea = False
-            jain = request.POST["jain"]
-            # if jain == "True":
-            #     high_tea = False
-            if caterer_list.count() < 1:
-                first_pref = None
-            else:
-                first_pref = request.POST["first_pref"]
-                caterer1 = Caterer.objects.get(name=first_pref)
-            if caterer_list.count() < 2:
-                second_pref = None
-            else:
-                second_pref = request.POST["second_pref"]
-                caterer2 = Caterer.objects.get(name=second_pref)
-            if caterer_list.count() < 3:
-                third_pref = None
+            jain = request.POST["jain"] == "True"
+            caterer_prefs = [
+                request.POST.get(pref) for pref in ["first_pref", "second_pref", "third_pref"]
+            ]
+            caterer_prefs = [Caterer.objects.get(name=pref) for pref in caterer_prefs if pref]
+
+            caterer = next((c for c in caterer_prefs if c.student_limit > Allocation.objects.filter(caterer=c, period=period_obj).count()), None)

```

Comment:
can we rewrite this to make it clear what we intend to do. Right now unreadable.

Reply:
tr breaking it into 2-3 lines